src/Workspaces/Core/Portable/Utilities/ParameterName.cs

```diff
@@ -36,6 +38,25 @@ public ParameterName(string nameBasedOnArgument, bool isFixed)
             }
         }
 
+        public ParameterName(string nameBasedOnArgument, bool isFixed, NamingRule parameterNamingRule)
+        {
+            NameBasedOnArgument = nameBasedOnArgument;
+
+            if (isFixed)
+            {
+                // If the parameter name is fixed, we have to accept it as is.
+                BestNameForParameter = NameBasedOnArgument;
+            }
+            else
+            {
+                // Otherwise, massage it a bit to be a more suitable match for
+                // how people actually writing parameters.
+                var trimmed = nameBasedOnArgument.TrimStart('_');
```

Comment:
💡 Would be good to trim the same set of prefixes implemented in #32549

Reply:
Added this issue #33673